<?xml version="1.0" ?>

<!--Name of the robot: pan_tilt-->
<robot name="iiwa7_robot">
    <!--First link: the base_link-->
    <link name="base_link">
        <!--Grahical shape -->
        <visual>
            <geometry>
                <cylinder length="0.17" radius="0.08"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.085"/>
            <material name="grey">
                <color rgba="0.5 0.5 0.5 1"/>
            </material>
        </visual>
        <!--Physical shape -->
        <collision>
            <geometry>
                <cylinder length="0.17" radius="0.08"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>

    <!--Joint connecting the base_link with the pan_link-->
    <joint name="joint1" type="revolute">
        <parent link="base_link"/>
        <child link="link1"/>
        <origin xyz="0 0 0.17"/>
        <axis xyz="0 0 1" />
        <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    </joint>

    <link name="link1">
        <visual>
            <geometry>
                <cylinder length="0.17" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.085"/>
            <material name="orange">
                <color rgba="0.95 0.61 0.07 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.17" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.085"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
        </inertial>
    </link>

    <joint name="joint2" type="revolute">
        <parent link="link1"/>
        <child link="link2"/>
        <origin rpy="1.57079632679  0 0" xyz="0 0 0.23"/>
        <axis xyz="0 0 1" />
        <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    </joint>

    <link name="link2">
        <visual>
            <geometry>
                <cylinder length="0.1" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="red">
                <color rgba="1 0 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.1" radius="0.06"/>
            </geometry>
            <origin rpy="0 1.57079632679 0" xyz="0 0 0"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
        </inertial>
    </link>

    <joint name="joint3" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin rpy="-1.57079632679 0 0" xyz="0 0.06 0"/>
        <axis xyz="0 0 1" />
        <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    </joint>

    <link name="link3">
        <visual>
            <geometry>
                <cylinder length="0.30" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.15"/>
            <material name="orange">
                <color rgba="0.95 0.61 0.07 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.30" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.15"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
        </inertial>
    </link>


    <joint name="joint4" type="revolute">
        <parent link="link3"/>
        <child link="link4"/>
        <origin rpy="1.57079632679  0 0" xyz="0 0 0.36"/>
        <axis xyz="0 0 1" />
        <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    </joint>

    <link name="link4">
        <visual>
            <geometry>
                <cylinder length="0.1" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="red">
                <color rgba="1 0 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.1" radius="0.06"/>
            </geometry>
            <origin rpy="0 1.57079632679 0" xyz="0 0 0"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
        </inertial>
    </link>


    <joint name="joint5" type="revolute">
        <parent link="link4"/>
        <child link="link5"/>
        <origin rpy="-1.57079632679 0 0" xyz="0 0.06 0"/>
        <axis xyz="0 0 1" />
        <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    </joint>

    <link name="link5">
        <visual>
            <geometry>
                <cylinder length="0.30" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.15"/>
            <material name="orange">
                <color rgba="0.95 0.61 0.07 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.30" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.15"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
        </inertial>
    </link>

    <joint name="joint6" type="revolute">
        <parent link="link5"/>
        <child link="link6"/>
        <origin rpy="1.57079632679  0 0" xyz="0 0 0.36"/>
        <axis xyz="0 0 1" />
        <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    </joint>

    <link name="link6">
        <visual>
            <geometry>
                <cylinder length="0.1" radius="0.06"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="red">
                <color rgba="1 0 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.1" radius="0.06"/>
            </geometry>
            <origin rpy="0 1.57079632679 0" xyz="0 0 0"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
        </inertial>
    </link>



    <joint name="joint7" type="revolute">
        <parent link="link6"/>
        <child link="end_effector"/>
        <origin rpy="-1.57079632679 0 0" xyz="0 0.06 0"/>
        <axis xyz="0 0 1" />
        <limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
    </joint>

    <link name="end_effector">
        <visual>
            <geometry>
                <box size="0.06 0.06 0.07"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.025"/>
            <material name="yellow">
                <color rgba="1 1 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <box size="0.06 0.06 0.07"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.025"/>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
        </inertial>
    </link>

</robot>
